### ISSUE TITLE
🚨 Security Vulnerability: ip npm package is unsafe for use as of v1.1.8

### ISSUE THREAD

**Issue by taylorjdawson** [react-native-community/cli#2294]
NPM `ip` package is vulnerable to Server-Side Request Forgery (SSRF) attacks, see [GitHub advisory](https://github.com/advisories/GHSA-78xj-cgh5-2h22) for more information.

 

> The impact of this vulnerability is significant, as it can lead to Server-Side Request Forgery (SSRF) attacks. SSRF attacks occur when an attacker is able to induce a server to make requests to internal resources, which the attacker normally couldn't access directly. In this case, the misclassification of 0x7f.1 as a public IP address allows for unintended internal network requests. This could potentially expose sensitive information, interact with internal services, or exploit vulnerabilities within the network. Additionally, such vulnerabilities can be leveraged to perform port scanning, gain unauthorized access, or as part of a larger attack chain. The risk is heightened in environments where the server has access to critical internal systems.
> — https://cosmosofcyberspace.github.io/npm_ip_cve/npm_ip_cve.html

**Comment by IzumiKomatsu** [comment]
Thanks, i'm facing the same issue.

**Comment by szymonrybczak** [comment]
hey @taylorjdawson, thanks for reporting! I've just created #2295 bumping `ip` package 👍

**Comment by taylorjdawson** [comment]
@szymonrybczak does that fix it though? The issue I believe is with the latest version of the `ip` package.

**Comment by szymonrybczak** [comment]
@taylorjdawson ah, right. I wrongly looked and I thought it was `<1.1.18`, let's see if maintainers will release a patch fix.

**Comment by DiNMEA** [comment]
so im not only one who faced that problem :/
anyone found an fix to this? i need to complete the tasks real bad lol

**Comment by glitch-txs** [comment]
seems like this upstreams to the latest version of react-native

**Comment by salmaazakii** [comment]
Hey, I'm facing the same issue, Should I downgrade my @react-native-community/cli package version to a certain one so it will not be affected by the latest upgrade?

this is the 'npm audit' result:
# npm audit report

ip  <=1.1.8
Severity: high
NPM IP package vulnerable to Server-Side Request Forgery (SSRF) attacks - https://github.com/advisories/GHSA-78xj-cgh5-2h22
fix available via `npm audit fix --force`
Will install react-native@0.68.7, which is a breaking change
node_modules/ip
  @react-native-community/cli-doctor  *
  Depends on vulnerable versions of ip
  node_modules/@react-native-community/cli-doctor
    @react-native-community/cli  >=4.13.0
    Depends on vulnerable versions of @react-native-community/cli-doctor
    Depends on vulnerable versions of @react-native-community/cli-hermes
    node_modules/@react-native-community/cli
      react-native  <=0.0.0-ffdfbbec0 || >=0.69.0-rc.0
      Depends on vulnerable versions of @react-native-community/cli
      node_modules/react-native
        react-native-pie-chart  >=3.0.0
        Depends on vulnerable versions of react-native
        node_modules/react-native-pie-chart
  @react-native-community/cli-hermes  *
  Depends on vulnerable versions of ip
  node_modules/@react-native-community/cli-hermes
  
  thank you in advance

**Comment by darakeon** [comment]
The ip package last update was 2 years ago. 
https://www.npmjs.com/package/ip

My problem is with puppeteer instead of react, but same issue: will really ip package be updated? I think it is not maintained anymore...

**Comment by jxia-innablr** [comment]
anyone find any workaround pls share here, thanks!

**Comment by glitch-txs** [comment]
> anyone find any workaround pls share here, thanks!

there's no workaround, the libary needs to be either patched or replaced

**Comment by thymikee** [comment]
FYI, the only affected command is `profile-hermes` when producing source maps. If you're not using it on a server (e.g. your CI), you're safe to ignore this and wait for us to patch it once we have a proper solution. If you are using it however, please disable it temporarily.

**Comment by henriquelomarques** [comment]
same error here

**Comment by antoinecaputo** [comment]
The PR is opened [here](https://github.com/indutny/node-ip/pull/138) and should probably be merged soon.

**Comment by mnikolaus** [comment]
> The PR is opened [here](https://github.com/indutny/node-ip/pull/138) and should probably be merged soon.

I wouldn't count on it to be merged soon... I've seen a lot of depended libraries moving away from `ip` lib. I think @taylorjdawson proposal is very sound

**Comment by whayu901** [comment]
> > anyone find any workaround pls share here, thanks!
> 
> there's no workaround, the library needs to be either patched or replaced

i have tried patched the ip library. But still not work for me

**Comment by lsmith77** [comment]
FYI there is now a 1.1.9 and 2.0.1.

However the CVE needs to be updated to allow the 1.1.9 release to be seen as a valid fix:
https://github.com/github/advisory-database/pull/3553

**Comment by thymikee** [comment]
1.1.9 is within the semver range so refreshing the lock file should be enough. Additionally we merged #2299 (which drops the dependency on `ip` offering an alternative) which we intend to backport to RN 0.72 and 0.71 – although with the fix being within the semver range, I'm not sure if that's gonna be necessary and will leave that to the RN release crew to decide.

**Comment by lsmith77** [comment]
yeah .. I just wanted to point out that this release now exists and that people might still see dependabot etc complain until the CVE is updated.

**Comment by thymikee** [comment]
Thanks for doing that @lsmith77

**Comment by azmainamin** [comment]
If using React Native version 0.68.2. How can we get the patch/fix?

**Comment by thymikee** [comment]
@azmainamin it should be enough to regenerate the lockfile entry for this package in your project, now that https://github.com/github/advisory-database/pull/3553 is merged
